### PR TITLE
integration-cli: cleanup some of the test-utilities

### PR DESCRIPTION
**Pull request by thaJeztah**
Had some of these changes stashed; lots more to be cleaned up (and remove abstraction layers that make things hard to follow), but let me just start with pushing these small changes.

### integration-cli: remove WaitRestart(), un-export WaitForInspectResult()

- Remove `WaitRestart()` as it was no longer used
- Un-export `WaitForInspectResult()` as it was only used internally, and we want to reduce uses of these utilities.
- Inline `appendDocker()` into `Docker()` as it was the only place it was used, and the name was incorrect anyway (should've been named `prependXX`).
- Simplify `Args()`, as it was first splitting the slice (into `command` and `args`),   only to join them again into a single slice (in `icmd.Command()`).

### integration-cli: remove cli.Build(), cli.Inspect()

They were just small wrappers arround cli.Args(), and the abstraction made one wonder if they were doing some "magic" things, but they weren't, so just inlining the `cli.Args()` makes it more transparent what's executed.

### CODE REVIEW

**Review by corhere — APPROVED** [review]

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -106,12 +106,9 @@ func (s *DockerDaemonSuite) TestDaemonRestartWithVolumesRefs(c *testing.T) {
...
 	}
 
-	out, err := s.d.Cmd("inspect", "-f", "{{json .Mounts}}", "volrestarttest1")
+	out, err := s.d.Cmd("inspect", "-f", `{{range .Mounts}}{{if eq .Destination "/foo"}}FOUND{{end}}{{end}}`, "volrestarttest1")
 	assert.NilError(c, err, out)
-
-	if _, err := inspectMountPointJSON(out, "/foo"); err != nil {
-		c.Fatalf("Expected volume to exist: /foo, error: %v\n", err)
-	}
+	assert.Equal(c, out, "FOUND")
```
Comment (corhere) [inline integration-cli/docker_cli_daemon_test.go:111]: Pretty sure this'll fix the test case.
```suggestion
	assert.Equal(c, strings.TrimSpace(out), "FOUND")
```

File: integration-cli/docker_cli_daemon_test.go [hunk 2]
```diff
@@ -106,12 +106,9 @@ func (s *DockerDaemonSuite) TestDaemonRestartWithVolumesRefs(c *testing.T) {
 		c.Fatal(err, out)
 	}
 
-	out, err := s.d.Cmd("inspect", "-f", "{{json .Mounts}}", "volrestarttest1")
+	out, err := s.d.Cmd("inspect", "-f", `{{range .Mounts}}{{if eq .Destination "/foo"}}FOUND{{end}}{{end}}`, "volrestarttest1")
 	assert.NilError(c, err, out)
```
Comment (corhere) [inline integration-cli/docker_cli_daemon_test.go:110]: It'd be handy if the test listed all the mounts that were present if the assertion fails.
```suggestion
	out, err := s.d.Cmd("inspect", "-f", `{{range .Mounts}}{{.Destination}}{{'\0'}}{{end}}`, "volrestarttest1")
	assert.NilError(c, err, out)
	assert.Contains(c, strings.Split(out, '\0'), "/foo")
```

**Review by corhere — COMMENTED** [review]
LGTM, though one of the refactored tests is currently broken.

File: integration-cli/docker_cli_daemon_test.go (hunk 1 above)
Comment (thaJeztah) [reply integration-cli/docker_cli_daemon_test.go:111]: oh! I saw it fail earlier, and was wondering if it was a flake or an actual issue; let me fix that one

File: integration-cli/docker_cli_daemon_test.go (hunk 2 above)
Comment (thaJeztah) [reply integration-cli/docker_cli_daemon_test.go:110]: Good idea. will add
